Lib/idlelib/idle_test/test_parenclose.py

```diff
@@ -0,0 +1,97 @@
+'''Test idlelib.parenclose.
+'''
+from idlelib.ParenClose import ParenClose
+import unittest
+# Can't use mock text; it doesn't handle positioning correctly.
+from tkinter import Text
+from test.support import requires
+requires('gui')
+
+class MockEvent(object):
+    def __init__(self, char):
+        self.char = char
+
+
+class ParenCloseTest(unittest.TestCase):
+    def test_parenClose(self):
+        p = ParenClose()
+        p.text = Text()
+        p_open = p.p_open_event
+        p_close = p.p_close_event
+        t_open = p.t_open_event
+        back = p.back_event
+        delete = p.delete_event
+        for paren_close_set, tick_close_set, skip_closures_set, mutual_b_set, \
+            mutual_d_set, insert, func, pos, char, result, posend in [
+              (1, 1, 1, 1, 1, 'def abuse', p_open, '1.9', '(',
```

Comment:
`for test_tuple in [<list of tuples>]` would allow test_tuple to be passed to a helper function.